plugin/pkg/scheduler/factory/factory.go

```diff
@@ -307,6 +515,64 @@ func (c *ConfigFactory) updateNodeInCache(oldObj, newObj interface{}) {
 	if err := c.schedulerCache.UpdateNode(oldNode, newNode); err != nil {
 		glog.Errorf("scheduler cache UpdateNode failed: %v", err)
 	}
+
+	c.invalidateCachedPredicatesOnNodeUpdate(newNode, oldNode)
+}
+
+func (c *ConfigFactory) invalidateCachedPredicatesOnNodeUpdate(newNode *v1.Node, oldNode *v1.Node) {
+	if c.enableEquivalenceClassCache {
+		// Begin to update equivalence cache based on node update
+		// TODO(resouer): think about lazily initialize this set
+		invalidPredicates := sets.NewString()
+
+		oldTaints, oldErr := helper.GetTaintsFromNodeAnnotations(oldNode.GetAnnotations())
+		if oldErr != nil {
+			glog.Errorf("Failed to get taints from old node annotation for equivalence cache")
+		}
+		newTaints, newErr := helper.GetTaintsFromNodeAnnotations(newNode.GetAnnotations())
+		if newErr != nil {
+			glog.Errorf("Failed to get taints from new node annotation for equivalence cache")
+		}
+
+		switch {
+		case !reflect.DeepEqual(oldNode.Status.Allocatable, newNode.Status.Allocatable):
+			invalidPredicates.Insert("GeneralPredicates") // "PodFitsResources"
+		case !reflect.DeepEqual(oldNode.GetLabels(), newNode.GetLabels()):
```

Comment:
I think that we generally should check all the cases here (e.g.. if the previous were true, this one should also be handled).
This is not how switch works - it breaks on the first matched case, so I think using switch here is incorrect.